dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportInstance.java

```diff
@@ -82,7 +81,7 @@ public void init(List<MetadataReportConfig> metadataReportConfigs) {
     private void init(MetadataReportConfig config, MetadataReportFactory metadataReportFactory) {
         URL url = config.toUrl();
         if (METADATA_REPORT_KEY.equals(url.getProtocol())) {
-            String protocol = url.getParameter(METADATA_REPORT_KEY, DEFAULT_DIRECTORY);
+            String protocol = url.getParameter(METADATA_REPORT_KEY, config.getProtocol());
```

Comment:
`url.getParameter(METADATA_REPORT_KEY, isEmpty(config.getProtocol()) ? dubbo :config.getProtocol() );`

I think we need to check if `config.getProtocol()` is empty here.

Reply:
I agree with you,but metadata report center support redis/zookeeper/nacos/…….Default value ```dubbo``` will occur a error when user‘s configuration is empty.
so,I think we need to check the config is empty.it throws exception for user to remind that they need to config metadata report center protocol correctly.

Reply:
If end users want to use metadata  report center correctly,they must config ```dubbo.metadata-report.address ```like ```protocol://ip:port```,otherwise,they must config ```dubbo.metadata-report.address```,```dubbo.metadata-report.port``` and ```dubbo.metadata-report.protocol ```.If we give a default value ```dubbo```(Of course it is wrong),then when properties isn't correct,the application cannot start up successfully and application throws exception like ```java.lang.IllegalStateException: No such extension org.apache.dubbo.metadata.report.MetadataReportFactory by name dubbo, no related exception was found, please check whether related SPI module is missing.```
But if we don't give default value,the application will start up successfully but the metadata report center cannot work
 So I think we need to print a error log better than give default value dubbo.

Reply:
I add an error log when the protocol cannot be parsed correctly.

Reply:
> so,I think we need to check the config is empty.it throws exception for user to remind that they need to config metadata report center protocol correctly.

Totally agree, that would be better.

Reply:
Please check this patch https://github.com/apache/dubbo/pull/9820

I think it's better to check protocol configuration on the construction of metadata config url.